packages/compass-aggregations/src/modules/aggregation.ts

```diff
@@ -39,32 +113,131 @@ export const runAggregation = (): ThunkAction<
   RootState,
   void,
   Actions
+> => {
+  return (dispatch) => dispatch(fetchAggregationData(1));
+};
+
+export const fetchPrevPage = (): ThunkAction<
+  void,
+  RootState,
+  void,
+  Actions
+> => {
+  return (dispatch, getState) => {
+    const {
+      aggregation: { page }
+    } = getState();
+    if (page <= 1) {
+      return;
+    }
+    dispatch(fetchAggregationData(page - 1));
+  };
+};
+
+export const fetchNextPage = (): ThunkAction<
+  void,
+  RootState,
+  void,
+  Actions
+> => {
+  return (dispatch, getState) => {
+    const {
+      aggregation: { isLast, page }
+    } = getState();
+    if (isLast) {
+      return;
+    }
+    dispatch(fetchAggregationData(page + 1));
+  };
+};
+
+export const cancelAggregation = (): ThunkAction<
+  void,
+  RootState,
+  void,
+  Actions
+> => {
+  return async (_dispatch, getState) => {
+    const {
+      dataService: { dataService },
+      aggregation: { abortController, session },
+    } = getState();
+    abortController?.abort();
+    if (session && dataService) {
+      try {
+        await dataService.killSessions(session);
+      } catch (e) {
+        log.warn(mongoLogId(1001000105), 'Aggregations', 'Attempting to kill the session failed');
+      }
+    }
+  };
+};
+
+const fetchAggregationData = (page: number): ThunkAction<
+  void,
+  RootState,
+  void,
+  Actions
 > => {
   return async (dispatch, getState) => {
     const {
       pipeline,
       namespace,
       maxTimeMS,
       collation,
-      dataService: { dataService }
+      dataService: { dataService },
+      aggregation: { limit },
     } = getState();
 
-    const stages = pipeline.map(generateStage);
-    const options: AggregateOptions = {
-      maxTimeMS: maxTimeMS || DEFAULT_MAX_TIME_MS,
-      allowDiskUse: true,
-      collation: collation || undefined,
-    };
-    if (dataService) {
-      const cursor = dataService.aggregate(
+    if (!dataService) {
+      return;
+    }
+
+    try {
+      const abortController = new AbortController();
+      const signal = abortController.signal;
+      const session = dataService.startSession('CRUD');
+      dispatch({
+        type: ActionTypes.AggregationStarted,
+        abortController,
+        session,
+      });
+
+      const stages = pipeline.map(generateStage);
```

Comment:
`generateStage` returns disabled stages as `{}` which breaks pipeline when you are trying to run it:

<img width="594" alt="image" src="https://user-images.githubusercontent.com/5036933/161221268-02c3f074-1474-4fac-9335-78f3b83046ff.png">